### PR TITLE
fix: uniform client API

### PR DESCRIPTION
**Pull request by alespour**
## Proposed Changes

* Repackaged `influx` -> `influxdb3`
* Uniform client API, mainly
  * `Configs` -> `ClientConfig`
    *  `HostUrl` -> `Host`
    *  `AuthToken` -> `Token`
  * `WriteParams` -> `WriteOptions`
    * `Consistency` removed (not supported by 3.x)
  * `Query` method has sibling method `QueryWithOptions` with additional parameter of `QueryOptions` type, which allows to specify database and query type (like eg. in Java client)
  * `WriteXXX` methods has variants with `WriteXXXWithOptions` with parameter of `WriteOptions` type, which allows to specify database and precision (like eg. in Java client)

## Checklist

- [x] CHANGELOG.md updated
- [x] Rebased/mergeable
- [x] A test has been added if appropriate
- [x] Tests pass
- [x] Commit messages are [conventional](https://www.conventionalcommits.org/en/v1.0.0/)
- [x] Sign [CLA](https://www.influxdata.com/legal/cla/) (if not already signed)

### CODE REVIEW

File: influxdb3/config.go [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+*/
+
+package influxdb3
+
+import (
+	"net/http"
+)
+
+// ClientConfig holds the parameters for creating a new client.
+// The only mandatory field is ServerURL. AuthToken is also important
```
Comment (bednar) [inline influxdb3/config.go:30]:
```suggestion
// The only mandatory field is ServerURL. Token is also important
```

File: influxdb3/error.go [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+*/
+
+package influxdb3
+
+import (
+	"fmt"
+)
+
+// ServerError holds InfluxDB server error info
+// ServerError represents an error returned from an InfluxDB API server.
```
Comment (bednar) [inline influxdb3/error.go:30]:
```suggestion
// ServerError represents an error returned from an InfluxDB API server.
```

**Review by bednar — CHANGES_REQUESTED** [review]
Thanks again for your PR 👍 
There are a few requirements that must be satisfied before we accept the PR:

1. Please update `README.md` to new naming

and also:

File: influxdb3/query.go [hunk 3]
```diff
@@ -1,100 +1,124 @@
...
+//   - database: The database to be used for InfluxDB operations.
+//   - query: The SQL query string to execute.
+//   - queryParams: Additional query parameters.
+//
+// Returns:
+//   - A custom iterator (*QueryIterator) that can also be used to get raw flightsql reader.
+//   - An error, if any.
+func (c *Client) Query(ctx context.Context, database string, query string, queryParams ...string) (*QueryIterator, error) {
+	return c.queryWithType(ctx, database, query, "sql", queryParams...)
+}
```
Comment (bednar) [inline influxdb3/query.go:94]: The `database` parameter should be optional and can be configured via `ClientConfig`. Maybe we can use something like:

```golang
func (c *Client) Query(ctx context.Context, query string, cfg ...influxdb3.QueryConfings) (*QueryIterator, error) {
	...
}

//
// This is pretty common use case => we can cover it via own function
//
func (c *Client) Query(ctx context.Context, query string, database string) (*QueryIterator, error) {
	...
}

type QueryConfings struct {
	Database string
	QueryParams string...
}
```

**Review by bednar — CHANGES_REQUESTED** [review]

File: influxdb3/query.go (hunk 3 above)
Comment (alespour) [reply influxdb3/query.go:94]: Go client seems to require more refactoring. Eg. there are also no `write`  methods that allow to specify precision, unlike in C# or Java.
Java client has `WriteOptions {database, precision, gzipThreshold}` and `QueryOptions {database, queryType}`  and `write` and `query` methods overloads with a parameter of these types. I guess  we should use the same approach in Go?

Comment (alespour) [reply influxdb3/query.go]: Right, Java :)

File: example/main.go [hunk 4]
```diff
@@ -62,13 +63,13 @@ func main() {
...
 	// Write point
-	err = client.WriteData(context.Background(), database, sensorData)
+	err = client.WriteData(context.Background(), sensorData)
 	if err != nil {
 		panic(err)
 	}
 	// Or write directly line protocol
 	line := fmt.Sprintf("stat,unit=temperature avg=%f,max=%f", 23.5, 45.0)
-	err = client.Write(context.Background(), database, []byte(line))
+	err = client.Write(context.Background(), []byte(line))
```
Comment (bednar) [inline example/main.go:72]: The query parts should be change also


```golang
	# iterator, err := client.Query(context.Background(), database, query) =>
        iterator, err := client.Query(context.Background(), query)
```

File: influxdb3/query.go [hunk 5]
```diff
@@ -0,0 +1,136 @@
...
+// Parameters:
+//   - ctx: The context.Context to use for the request.
+//   - options: Query options (query type, optional database).
+//   - query: The query string to execute.
+//   - params: Additional query parameters.
+//
+// Returns:
+//   - A custom iterator (*QueryIterator) that can also be used to get raw flightsql reader.
+//   - An error, if any.
+func (c *Client) QueryWithOptions(ctx context.Context, options *QueryOptions, query string, params ...string) (*QueryIterator, error) {
```
Comment (bednar) [inline influxdb3/query.go:92]: I think that the `params` should be part of `QueryOptions`:

```suggestion
func (c *Client) QueryWithOptions(ctx context.Context, options *QueryOptions, query string) (*QueryIterator, error) {
```

**Review by bednar — CHANGES_REQUESTED** [review]
Thanks again for your PR 👍 
There are a few requirements that must be satisfied before we accept the PR:

File: example/main.go (hunk 4 above)
Comment (alespour) [reply example/main.go:72]: Fixed.

File: influxdb3/query.go (hunk 5 above)
Comment (bednar) [reply influxdb3/query.go]: ok

**Review by bednar — APPROVED** [review]
LGTM 🍻

**Pull request merged**